client/src/containers/ProtectedRoute/ProtectedRoute.js

```diff
@@ -14,26 +14,31 @@ class ProtectedRoute extends React.Component {
     }
 
     render() {
-        const { session, oneOf, component: Component, path, history } = this.props;
+        const { session, oneOf, component: Component, path, history, exact, customProps } = this.props;
 
         // Wait until we've finished getting the user session before making any redirects
         if (session.pending) {
             return (<></>);
         }
 
-        const noAccessRedirect = (<Redirect to={{
+        const noAccessRedirect = message => (<Redirect to={{
             pathname: '/',
             state: history.location.state && history.location.state.suppressAuthenticationBanner ? {} : {
-                alert: 'You do not have sufficient privileges to view that page.'
+                alert: message
             }
         }}/>);
 
         let loggedIn = session.user && session.user.role;
-        let render = (props) => (<Component {...props} />);
+        let render = (props) => (<Component {...props} {...customProps} />);
+
+        // If using exact path, match exactly, otherwise do some fuzzy checking
+        // TODO: Verify that this fuzzy checking is good enough
```

Comment:
Is it good enough?

Reply:
Seems to be... it basically allows `/monuments/:monumentId` for example to match to `/monuments/3`. I'm wondering if there's a good way to 100% verify that they match, maybe through react router...